packages/frontend/src/home/home.js

```diff
@@ -89,41 +88,33 @@ function mergePeriods(period0, period1) {
   return sumPackages;
 }
 
-function maxDate(a, b) {
-  return new Date(Math.max(a.getTime(), b.getTime()));
-}
-
 /**
  * @param names    {string} Package names
  * @param startDay {number} Start of period, 1 is yesterday
  * @param endDay   {number} End of period 0 is today
  * @returns {Promise<any>}
  */
-function getPackagesDownloadsOverPeriod(names, startDay, endDay) {
-  const requestPeriod = Math.min(maxRequestPeriod, startDay - endDay);
-  const requestEndDay = startDay - requestPeriod;
-
-  const startStats = new Date(Date.UTC(2015, 1, 10));
-
-  let startDate = maxDate(startStats, subDays(new Date(), startDay));
-  let endDate = maxDate(startStats, subDays(new Date(), requestEndDay));
-
-  startDate = formatDate(startDate, 'YYYY-MM-DD', null, 'UTC');
-  endDate = formatDate(endDate, 'YYYY-MM-DD', null, 'UTC');
-
-  const period1 = getPackagesDownloads(names, { startDate, endDate });
+async function getPackagesDownloadsOverPeriod(names, startDay, endDay) {
+  const requestPeriods = getPackageRequestPeriods(startDay, endDay);
+  const promises = requestPeriods.map(period => {
+    return getPackagesDownloads(names, {
+      startDate: period.startDate,
+      endDate: period.endDate,
+    });
+  });
+  const periods = await Promise.all(promises);
```

Comment:
Previously, I believe that if there are >2 promises, they are resolved sequentially. But now since there is a helper function to retrieve all the start/end dates, it is done in parallel